File: LuckyBol/app/src/main/java/com/pplus/luckybol/apps/common/mgmt/CategoryFirstManager.java
package com.pplus.luckybol.apps.common.mgmt;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.pplus.luckybol.LuckyBolApplication;
import com.pplus.luckybol.core.network.model.dto.CategoryFirst;
import com.pplus.utils.part.pref.PreferenceUtil;

import java.lang.reflect.Type;
import java.util.List;

/**
 * Created by 김종경 on 2016-10-06.
 */

public class CategoryFirstManager {

    private List<CategoryFirst> categoryFirstList;

    public List<CategoryFirst> getCategoryFirstList() {
        return categoryFirstList;
    }

    public void setCategoryFirstList(List<CategoryFirst> categoryFirstList) {
        this.categoryFirstList = categoryFirstList;
    }

    // Preference
    public static final String SHARED_CATEGORY_FIRST= "category_first";

    private static CategoryFirstManager mCategoryFirstManager = null;

    private CategoryFirstManager(){

    }

    public static CategoryFirstManager getInstance(){

        if(mCategoryFirstManager == null) {
            mCategoryFirstManager = new CategoryFirstManager();
            mCategoryFirstManager.load();
        }
        return mCategoryFirstManager;
    }

    public void load(){

        String data = PreferenceUtil.getDefaultPreference(LuckyBolApplication.getContext()).getString(SHARED_CATEGORY_FIRST);
        Type typeOfSrc = new TypeToken<CategoryFirstManager>(){

        }.getType();

        CategoryFirstManager categoryInfoManager = new Gson().fromJson(data, typeOfSrc);

        if(categoryInfoManager != null) {
            mCategoryFirstManager = new Gson().fromJson(data, typeOfSrc);
        }
    }

    public void save(){

        Type typeOfSrc = new TypeToken<CategoryFirstManager>(){

        }.getType();
        String data = new Gson().toJson(mCategoryFirstManager, typeOfSrc);
        PreferenceUtil.getDefaultPreference(LuckyBolApplication.getContext()).put(SHARED_CATEGORY_FIRST, data);
    }

    public void clear(){

        //PreferenceUtil.getPreference(mContext, SHARED_CATEGORY_INFO).clear();
        mCategoryFirstManager = new CategoryFirstManager();
        mCategoryFirstManager.setCategoryFirstList(null);
        save();
    }

    public boolean isCalled(){

        if(mCategoryFirstManager.getCategoryFirstList() != null) {
            return true;
        }

        return false;
    }
}

File: PRNumber/app/src/main/java/com/pplus/prnumberuser/apps/common/mgmt/CategoryFirstManager.kt
package com.pplus.prnumberuser.apps.common.mgmt

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.pplus.utils.part.pref.PreferenceUtil
import com.pplus.prnumberuser.PRNumberApplication
import com.pplus.prnumberuser.core.network.model.dto.CategoryFirst

/**
 * Created by 김종경 on 2016-10-06.
 */
class CategoryFirstManager private constructor() {
    var categoryFirstList: List<CategoryFirst>? = null

    fun load() {
        val data = PreferenceUtil.getDefaultPreference(PRNumberApplication.getContext()).getString(SHARED_CATEGORY_FIRST)
        val typeOfSrc = object : TypeToken<CategoryFirstManager?>() {}.type
        val categoryInfoManager = Gson().fromJson<CategoryFirstManager>(data, typeOfSrc)
        if (categoryInfoManager != null) {
            mCategoryFirstManager = Gson().fromJson(data, typeOfSrc)
        }
    }

    fun save() {
        val typeOfSrc = object : TypeToken<CategoryFirstManager?>() {}.type
        val data = Gson().toJson(mCategoryFirstManager, typeOfSrc)
        PreferenceUtil.getDefaultPreference(PRNumberApplication.getContext()).put(SHARED_CATEGORY_FIRST, data)
    }

    fun clear() {

        //PreferenceUtil.getPreference(mContext, SHARED_CATEGORY_INFO).clear();
        mCategoryFirstManager = CategoryFirstManager()
        mCategoryFirstManager!!.categoryFirstList = null
        save()
    }

    fun isCalled(): Boolean {
        return mCategoryFirstManager!!.categoryFirstList != null
    }

    companion object {
        // Preference
        const val SHARED_CATEGORY_FIRST = "category_first"
        private var mCategoryFirstManager: CategoryFirstManager? = null
        fun getInstance(): CategoryFirstManager{
            if (mCategoryFirstManager == null) {
                mCategoryFirstManager = CategoryFirstManager()
                mCategoryFirstManager!!.load()
            }
            return mCategoryFirstManager!!
        }
    }
}